src/read_distribution.rs

```diff
@@ -17,6 +17,45 @@ fn filename_from_file(path: impl AsRef<Path>) -> Result<String> {
         .to_string())
 }
 
+/// Standard Python wheel filename components (tags)
+///
+/// The wheel filename is "<name>-<version>-<python_tag>-<abi_tag>-<platform_tag>.whl"
+struct WheelFilenameParts<'a> {
```

Comment:
Are you sure using string slices is not OK here? That's what the original version of this code did as well, because these values are only needed temporarily, and most of the components are not even used. Also this is a private type and it can be easily changed to an owned version should the need arise.

Reply:
Slices or `&str` are not an issue, but explicit lifetimes add a huge amount of complexity. I'd rather like to remove them now than pass that task on to the next person that tries to refactor this code.

Reply:
All right, I've changed everything to `String`.